### PR TITLE
Add custom headers to email

### PR DESCRIPTION
**Pull request by ted-mey**
This a continuation on an existing PR #871 , and tries to handle the comments there. Also adding the property before the `restrict_keys` part to not sidestep that functionality.

We want to be able to send custom headers to the email, as supported in [this example i MS docs](https://learn.microsoft.com/en-us/graph/api/user-post-messages?view=graph-rest-1.0&tabs=http#example-2-create-message-draft-that-includes-custom-message-headers). 

I added two unit tests to verify the behavior ~but I wasn't able to test it with the live environment~. I've tested it with my O365-account and I think it works as expected.

### CODE REVIEW

**Comment by ted-mey** [comment]
@alejcas what do you think?

**Comment by alejcas** [comment]
> @alejcas what do you think?

Well I commented your code but you didn’t answer

**Comment by ted-mey** [comment]
@alejcas what do you think?

> image

Thank you! I've updated the code with your suggestion to fix the issue.

The reason why I can't see your comment is because it's "Pending", you need to submit it for me to be able to see it.  [Here is a github issue](https://github.com/orgs/community/discussions/10369) with lots of people confused about the UX.

**Comment by alejcas** [comment]
> @alejcas what do you think?
> 
> > image
> 
> Thank you! I've updated the code with your suggestion to fix the issue.
> 
> The reason why I can't see your comment is because it's "Pending", you need to submit it for me to be able to see it. [Here is a github issue](https://github.com/orgs/community/discussions/10369) with lots of people confused about the UX.

Embarrasing...

Review submited

Please see the comments

File: O365/message.py [hunk 1]
```diff
@@ -622,6 +622,20 @@ def single_value_extended_properties(self):
...
         return self.__single_value_extended_properties
 
+    @property
+    def message_headers(self):
+        """ Custom message headers
+            List of internetMessageHeaders, see definition: https://learn.microsoft.com/en-us/graph/api/resources/internetmessageheader?view=graph-rest-1.0
+        :type: list[dict[str, str]]
+        """
+
+        return self.__message_headers
```
[the remark this answers is not in the document]
Comment (ted-mey) [reply O365/message.py:632]: Thanks, done.

File: O365/message.py [hunk 2]
```diff
@@ -622,6 +622,20 @@ def single_value_extended_properties(self):
...
+
+    @message_headers.setter
+    def message_headers(self, value):
+        self.__message_headers = value
```
[the remark this answers is not in the document]
Comment (ted-mey) [reply O365/message.py:639]: Thanks, good input. I added a check to make sure the setter receives a list. To help out with the format and the need to just add a single header I instead added a new method `add_message_header`, I think that's even more helpful.

**Comment by alejcas** [comment]
Thanks!

**Comment by ted-mey** [comment]
> Thanks!

Thank you! What is the schedule of releases of this package? I noticed the last version is from februari on pypi. We would love to be able to use this in our application, when do you think it will be available?

**Comment by alejcas** [comment]
I usually wait until there are a coupe/more improvements on master.

Let me check and I’ll report back

**Comment by alejcas** [comment]
2.0.35 released

**Pull request merged**